src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets

```diff
@@ -65,6 +65,9 @@ Copyright (c) .NET Foundation. All rights reserved.
                  (
                  '$(RuntimeIdentifier)' == '' and
                  '$(RuntimeIdentifiers)' == '' and
+                 '$(_IsExecutable)' == 'true' and '$(IsTestProject)' != 'true' and
+                 '$(IsRidAgnostic)' != 'true' and
+                 '$(DisableImplicitRuntimeIdentifier)' != 'true' and
```

Comment:
Hm, I didn't think of that. That's a much better way to implement this. Thanks 😄

Reply:
I don't think this is the best approach anymore.

If we do this we will make all apps rid-specific by default because if they dont specify a RID we will imply one. The purpose of the old change was to make apps that already required a RID get a RID automatically.

And if we condition the change such that its UCR OR we set UCR and have the required properties, we still need to add another property to allow opting out if we don't want to break UCR. Like

```xml
<PropertyGroup Condition="'$(UseCurrentRuntimeIdentifier)' == ''">
    <ImpliedUCR>Condition="
                 '$(RuntimeIdentifier)' == '' and
                 '$(RuntimeIdentifiers)' == '' and
                 '$(_IsExecutable)' == 'true' and '$(IsTestProject)' != 'true' and
                 '$(IsRidAgnostic)' != 'true' ">true</ImpliedUCR>
           <UseCurrentRuntimeIdentifier Condition=" '$(ImpliedUCR) == 'true'">true</UseCurrentRuntimeIdenitifer>
</PropertyGroup>

<PropertyGroup Condition="'$(UseCurrentRuntimeIdentifier)' == 'true' and '$(ImpliedUCR)' != true or ('$(ImpliedUCR)' == 'true' and '$(UseCurrentRuntimeIdenitifer') == 'true' and 'SC' == true or 'PublishAOT' == true or 'PublishRR' == 'true' or ...)>
    <RuntimeIdentifier>$(NETCoreSdkPortableRuntimeIdentifier)</RuntimeIdentifier>
</PropertyGroup>
```

Doing the change this way would be a much higher risk change and may muddle the concept of UCR IMO. Thoughts @dsplaisted ?

Reply:
@nagilson Sorry, I didn't mean to change the behavior for default apps (though I think that's on the table for .NET 8).

I think we can probably use the same `UseCurrentRuntimeIdentifier ` though.  This is probably what I wanted in the original suggestion:

```xml
<PropertyGroup Condition="'$(UseCurrentRuntimeIdentifier)' == ''">
    <UseCurrentRuntimeIdentifier Condition="
                 '$(RuntimeIdentifier)' == '' and
                 '$(RuntimeIdentifiers)' == '' and
                 '$(_IsExecutable)' == 'true' and '$(IsTestProject)' != 'true' and
                 '$(IsRidAgnostic)' != 'true' and
                  (
                     '$(SelfContained)' == 'true' or
                     '$(PublishReadyToRun)' == 'true' or
                     '$(PublishSingleFile)' == 'true' or
                     '$(PublishAot)' == 'true'
                  )">true</UseCurrentRuntimeIdentifier>
</PropertyGroup>
<PropertyGroup Condition="'$(UseCurrentRuntimeIdentifier)' == 'true'>
    <RuntimeIdentifier>$(NETCoreSdkPortableRuntimeIdentifier)</RuntimeIdentifier>
</PropertyGroup>
```